dnserver/main.py

```diff
@@ -85,33 +85,59 @@ def __str__(self):
         return str(self.rr)
 
 
-class Resolver(ProxyResolver):
-    def __init__(self, zones_file: str | Path, upstream: str):
+def resolve(request, handler, records):
+    type_name = QTYPE[request.q.qtype]
+    reply = request.reply()
+    for record in records:
+        if record.match(request.q):
+            reply.add_answer(record.rr)
+
+    if reply.rr:
+        logger.info('found zone for %s[%s], %d replies', request.q.qname, type_name, len(reply.rr))
+        return reply
+
+    # no direct zone so look for an SOA record for a higher level zone
+    for record in records:
+        if record.sub_match(request.q):
+            reply.add_answer(record.rr)
+
+    if reply.rr:
+        logger.info('found higher level SOA resource for %s[%s]', request.q.qname, type_name)
+        return reply
+
+
+class BaseResolver(LibBaseResolver):
+    def __init__(self, zones_file: str | Path):
         records = load_records(zones_file)
         self.records = [Record(zone) for zone in records.zones]
-        logger.info('loaded %d zone record from %s', len(self.records), zones_file)
-        super().__init__(address=upstream, port=53, timeout=5)
+        logger.info('loaded %d zone record from %s, without proxy DNS server', len(self.records), zones_file)
+        super().__init__()
 
     def resolve(self, request, handler):
+        answer = resolve(request, handler, self.records)
+        if answer:
+            return answer
+
         type_name = QTYPE[request.q.qtype]
-        reply = request.reply()
-        for record in self.records:
-            if record.match(request.q):
-                reply.add_answer(record.rr)
+        logger.info('no local zone found, not proxying %s[%s]', request.q.qname, type_name)
+        return request.reply()
 
-        if reply.rr:
-            logger.info('found zone for %s[%s], %d replies', request.q.qname, type_name, len(reply.rr))
-            return reply
 
-        # no direct zone so look for an SOA record for a higher level zone
-        for record in self.records:
-            if record.sub_match(request.q):
-                reply.add_answer(record.rr)
+class ProxyResolver(LibProxyResolver):
+    def __init__(self, zones_file: str | Path, upstream: str):
+        records = load_records(zones_file)
+        self.records = [Record(zone) for zone in records.zones]
+        logger.info(
+            'loaded %d zone record from %s, with %s as a proxy DNS server', len(self.records), zones_file, upstream
+        )
+        super().__init__(address=upstream, port=DEFAULT_PORT, timeout=5)
```

Comment:
```suggestion
        super().__init__(address=upstream, port=53, timeout=5)
```
In case someone manually does

```py
import dnserver.main
dnserver.main.DEFAULT_PORT = 5053
```
Which is valid, if undocumented.